Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorContextMenu.cpp

```diff
@@ -0,0 +1,226 @@
+#include "Spire/OrderImbalanceIndicator/OrderImbalanceIndicatorContextMenu.hpp"
+#include <QEvent>
+#include <QWidgetAction>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Ui/CheckBox.hpp"
+#include "Spire/Ui/DropShadow.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+
+OrderImbalanceIndicatorContextMenu::OrderImbalanceIndicatorContextMenu(
+    QWidget* parent)
+    : QMenu(parent) {
+  setWindowFlag(Qt::NoDropShadowWindowHint);
+  DropShadow context_menu_shadow(true, true, this);
+  setFixedSize(scale(140, 90));
+  setStyleSheet(QString(R"(
+    QMenu {
+      background-color: #FFFFFF;
+      border: %1px solid #A0A0A0 %2px solid #A0A0A0;
+      color: #000000;
+      font-family: Roboto;
+      font-size: %3px;
+      padding: %4px 0px;
+    }
+
+    QMenu::item {
+      padding: %5px 0px %5px %6px;
+    }
+
+    QMenu::item:disabled,
+    QMenu::item:disabled:selected,
+    QMenu::item:disabled:hover {
+      background-color: #FFFFFF;
+      color: #C8C8C8;
+    }
+
+    QMenu::right-arrow {
+      height: %7px;
+      image: url(:/Icons/arrow-right-black.svg);
+      padding-right: %6px;
+      width: %8px;
+    }
+
+    QMenu::right-arrow:selected {
+      height: %7px;
+      image: url(:/Icons/arrow-right-white.svg);
+      padding-right: %6px;
+      width: %8px;
+    }
+
+    QMenu::item:selected, QMenu::item:hover {
+      background-color: #8D78EC;
+      color: #FFFFFF;
+    })")
+    .arg(scale_height(1)).arg(scale_width(1))
+    .arg(scale_height(12)).arg(scale_height(5))
+    .arg(scale_height(3)).arg(scale_width(8))
+    .arg(scale_height(6)).arg(scale_width(4)));
+  m_table_columns_menu = new QMenu(tr("Table Columns"), this);
+  m_table_columns_menu->installEventFilter(this);
+  m_table_columns_menu->setFixedSize(scale(140, 150));
+  m_table_columns_menu->setContentsMargins(0, scale_height(0), 0,
+    scale_height(4));
+  addMenu(m_table_columns_menu);
+  auto security_action = new QWidgetAction(this);
```

Comment:
For all the actions, use `std::unique_ptr` and call `action.get()` where you need a raw pointer handle without ownership transfer and `action.release()` to transfer ownership.
`std::unique_ptr` provides exception safety and incurs no runtime overhead. As a side effect, in the modern C++ world, in makes the intent clearer regarding whether the object is borrowed or acquired.

Reply:
On the subject of ownership, the ```this``` being passed to the ```QWidgetAction``` constructor establishes ownership, namely that ```this``` owns ```security_action``` so using a ```unique_ptr``` to also establish ownership as well ends up being redundant and error prone.